### PR TITLE
Add Tags to blog

### LINKED ISSUE #3021: Add tags to each blog post
### Description

Instead of linking the most recent articles in `/blog`, we should add tags to each blog post and display the three most relevant ones. Related to https://github.com/gitpod-io/website/issues/1683

### PR DESCRIPTION
**Pull request by Cahllagerfeld**
closes #3021

For better UX, SSR gets enabled for the `/blog` route. otherwise the filtering would happen `onMount` which looks a bit weird to the user

Added a new prompt to the `npm run generate:blog-post` when creating a new post

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3239"></a>

### CODE REVIEW

File: src/app.d.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ declare global {
       guides: any[];
       securityLogs: any[];
       educationCustomers: any[];
+      filter: string;
```
Comment (Siddhant-K-code) [inline src/app.d.ts:21]: **(early) suggestion:** can we use something as `tag` or `category`? wdyt?

URL Slug would be something like => `/blog?tag=Engineering`

**Review by Cahllagerfeld — COMMENTED** [review]
Hey @chrifro I added the tags from notion to the corresponding blog posts. There are still a lot of posts missing without one. Do we need to add them within this PR, or shall we do this as a follow up if required?

**Comment by chrifro** [comment]
Hi @Cahllagerfeld , the ones mentioned in the RFC are just some examples. Can you add the tags to all other blogs post? If you are uncertain about any, you could make a table overview for them and we figure it out together. I'd say all blog posts since 2021 should have a tag.

**Review by chrifro — COMMENTED** [review]
Awesome @Cahllagerfeld  ✨


### **Some comments:**
**on/blog**

- [x]  the hover for the filters is missing

Bildschirm­foto 2023-01-16 um 09 17 17

- [x]  I used the [secondary button style](https://www.figma.com/file/VL0CJpHqcDJVJ4BEkoyRFV/Gitpod-Website-Component-Library?node-id=513%3A527&t=tWnkDc3sL5udmUrq-1) which has a defined height of 32 px. Your button is 37px in height. What's your intention behind that?

- [ ] on dark theme: the label shadow is off. Can also be addressed in #1950 [see [component library](https://www.figma.com/file/VL0CJpHqcDJVJ4BEkoyRFV/Gitpod-Website-Component-Library?node-id=557%3A1088&t=tWnkDc3sL5udmUrq-1)]

Bildschirm­foto 2023-01-16 um 09 24 38
 
 **On /blog/xyz**:

- [ ] when being on a blog post about "engineering" the "you might also like section" should only display other blog posts with the same label. Let's display 3 random blog posts with the same label from the last 12 months. 
 
 
 - how can we make sure that a new tag will be added for new blog posts? Any ideas? 
 
unrelated: 
 - the minimum spacing between text and authors should be 32 px
Bildschirm­foto 2023-01-16 um 09 30 14

**Comment by Cahllagerfeld** [comment]
I'm a bit on the fence with picking random blogposts to display because the result won't be predictable. We may link three different ones after rebuilding the page, which would lead to inconsistencies also with SEO, because the linked pages now link to other pages.

I don't know if this is a bad impact or not though.
Ideally we would have a way to have 3 different posts per post, but in a predictable way I think 🤔

**Comment by Cahllagerfeld** [comment]
- added a prng for being able to predict the result of the random generate for the blogposts displayed in the section
- updated the button to use the actual button component
- updated spacing for avatars

**Review by chrifro — APPROVED** [review]
Works great! Awesome solution for displaying the same random blog posts ✨

Let's merge it 🚀

**Pull request merged**